### ISSUE TITLE
Requests for .js files respond with 422 Unprocessable Entity

### ISSUE THREAD

**Issue by chrisroos** [freerange/site#50]
I came across this while investigating an exception captured in Errbit - https://gofreerange-errbit.herokuapp.com/apps/54e2340aed53ab1ae5000002/problems/596f062c9c25a400064f4c16.

It seems that requests for paths with .js file extensions respond with 422 Unprocessable Entity. I think I'd expect them to respond with 404 given that the path/resource doesn't exist.

Steps to replicate the problem:

```
$ curl http://gofreerange.com/foo.js -s -v > /dev/null
*   Trying 85.159.213.6...
* Connected to gofreerange.com (85.159.213.6) port 80 (#0)
> GET /foo.js HTTP/1.1
> Host: gofreerange.com
> User-Agent: curl/7.43.0
> Accept: */*
> 
< HTTP/1.1 422 Unprocessable Entity
< Date: Wed, 19 Jul 2017 10:25:14 GMT
< Server: Apache/2.4.7 (Ubuntu)
< X-Runtime: 0.024108
< X-Request-Id: 5c996065-465a-4c95-b67a-4444aaaf4f01
< X-Powered-By: Phusion Passenger 5.0.30
< Content-Length: 1547
< Status: 422 Unprocessable Entity
< Cache-Control: max-age=604800
< Expires: Wed, 26 Jul 2017 10:25:14 GMT
< Content-Type: text/html; charset=utf-8
< 
{ [1063 bytes data]
* Connection #0 to host gofreerange.com left intact
```

**Comment by floehopper** [comment]
This 422 response appears to be generated by Rails [raising a `ActionController::InvalidCrossOriginRequest`](https://github.com/rails/rails/blob/v5.0.3/actionpack/lib/action_controller/metal/request_forgery_protection.rb#L239).

**Comment by floehopper** [comment]
This was fixed by 3364bd2c2949f5a1b3c26e28d708daad9319d241. We now see a "404 Not Found" response as expected:

```
$ curl http://gofreerange.com/foo.js -s -v > /dev/null
*   Trying 85.159.213.6...
* Connected to gofreerange.com (85.159.213.6) port 80 (#0)
> GET /foo.js HTTP/1.1
> Host: gofreerange.com
> User-Agent: curl/7.43.0
> Accept: */*
> 
< HTTP/1.1 404 Not Found
< Date: Thu, 20 Jul 2017 07:49:41 GMT
< Server: Apache/2.4.7 (Ubuntu)
< Cache-Control: no-cache
< X-Runtime: 0.008102
< X-Frame-Options: SAMEORIGIN
< X-XSS-Protection: 1; mode=block
< X-Content-Type-Options: nosniff
< X-Request-Id: 404e62b1-d59c-41bf-82a8-d3e5c043dcbb
< X-Powered-By: Phusion Passenger 5.0.30
< Status: 404 Not Found
< Cache-Control: max-age=604800
< Expires: Thu, 27 Jul 2017 07:49:41 GMT
< Transfer-Encoding: chunked
< Content-Type: text/plain; charset=utf-8
< 
{ [25 bytes data]
* Connection #0 to host gofreerange.com left intact
```

Closing.
